rag/llm/sequence2txt_model.py

```
@@ -512,19 +512,56 @@ def transcription(self, audio, max_retries=60, retry_interval=5):
+        headers = {"Authorization": f"Bearer {self.key}"} if self.key else {}
+
+        try:
+            response = requests.post(url=append_api_path(self.base_url, "audio/transcriptions"), files=files, data=payload, headers=headers, timeout=60)
+            response.raise_for_status()
+            result = response.json()
+            text = (result.get("text") or "").strip()
+            if text:
+                return text, num_tokens_from_string(text)
+            return "**ERROR**: Failed to retrieve transcription.", 0
+        except requests.exceptions.RequestException as e:
+            return f"**ERROR**: {str(e)}", 0
```

Ruff (0.16.1)

[warning] Use explicit conversion flag

Replace with conversion flag

(RUF010)